src/core/execute.c

```diff
@@ -7442,6 +7442,41 @@ void exec_shared_runtime_vacuum(Manager *m) {
         }
 }
 
+int exec_runtime_make(ExecSharedRuntime *shared, ExecRuntime **ret) {
+        _cleanup_(exec_runtime_freep) ExecRuntime *rt = NULL;
+
+        assert(ret);
+
+        if (!shared) {
+                *ret = NULL;
+                return 0;
+        }
+
+        rt = new(ExecRuntime, 1);
+        if (!rt)
+                return -ENOMEM;
+
+        *rt = (ExecRuntime) {
+                .shared = shared,
+        };
+
+        *ret = TAKE_PTR(rt);
+        return 1;
+}
+
+ExecRuntime* exec_runtime_free(ExecRuntime *rt, bool destroy) {
+        if (!rt)
+                return NULL;
+
+        exec_shared_runtime_unref(rt->shared, destroy);
+        return mfree(rt);
+}
```

Comment:
hmm, please split this into two functions: 

exec_runtime_free() and exec_runtime_destroy() or so.

we used to do things like that more often, where a boolean param would indiciate the "level of destruction" intended. But I think it's a bad style, and instead we should have always separate functions for that, at least in exported (non-static) APIs, and just have the tougher destructors do their thing and then call the weaker destructors as part of what they do.

I think ultimately that leads to much cleaner code because the dangerous and the non-dangerous destruction code are clearly separated out.